src/SignalR/server/Core/src/HubConnectionContext.cs

```diff
@@ -116,7 +116,7 @@ internal StreamTracker StreamTracker
         /// <summary>
         /// Gets the user for this connection.
         /// </summary>
-        public virtual ClaimsPrincipal? User => Features.Get<IConnectionUserFeature>()?.User;
+        public virtual ClaimsPrincipal User => Features.Get<IConnectionUserFeature>()?.User ?? new ClaimsPrincipal();
```

Comment:
Not sure what to do here. Cache the `User`?

The reason I'm not keeping it null is because it's nicer to deal with an empty `ClaimsPrincipal` than a null one. An empty one will not be authenticated in any way or have any claims. The Auth types we use don't have null annotations for the user as the expectation is that you'll at least pass an empty one.

Reply:
Assign blank `ClaimsPrincipal` to underlying `IConnectionUserFeature` when the property is accessed?

Reply:
The `IConnectionUserFeature` could come from anyone (in theory), and the interface allows a null user. Plus the interface itself could not be on the connection.